### DISCUSSION TITLE
Updates with composite primary key

### DISCUSSION THREAD

**Discussion by adamhp** [diesel-rs/diesel#3701] [Q&A] [answered]
I'm attempting to perform an update with a composite primary key using "AsChangeset" (and other methods). This composite primary key is actually two foreign keys. I'm attempting to [follow the documentation ](https://diesel.rs/guides/all-about-updates.html) using "AsChangeset", but each time I attempt to perform an update query, I get an error saying `no such column: id`. I've tried basically every possible permutation of the update docs, so I'm thinking it has something to do with the composite primary key? Would really appreciate any ideas here.

## The query
```rust
diesel::update(scores).set(score_form).execute(conn);
```

`score_form` is getting deserialized and should look something like:
```json
{
   "assessment_id": 1,
   "criterion_id": 2,
   "value": 5
}
```

## Models
```rust
#[derive(Identifiable, Insertable, AsChangeset, Deserialize)]
#[diesel(table_name = scores)]
#[diesel(primary_key(assessment_id, criterion_id))]
pub struct ScoreForm {
  pub assessment_id: i32,
  pub criterion_id: i32,
  pub dtm_created: Option<NaiveDateTime>,
  pub dtm_modified: Option<NaiveDateTime>,
  pub value: Option<i32>
}

#[derive(Identifiable, Queryable, Selectable, Serialize, Associations, PartialEq, Debug)]
#[diesel(table_name = scores)]
#[diesel(primary_key(assessment_id, criterion_id))]
#[diesel(belongs_to(Criterion))]
#[diesel(belongs_to(Assessment))]
#[diesel(check_for_backend(diesel::sqlite::Sqlite))]
pub struct Score {
  pub assessment_id: i32,
  pub criterion_id: i32,
  pub dtm_created: NaiveDateTime,
  pub dtm_modified: NaiveDateTime,
  pub value: Option<i32>
}
```

## Schema
```rust
diesel::table! {
    scores (assessment_id, criterion_id) {
        assessment_id -> Integer,
        criterion_id -> Integer,
        dtm_created -> Timestamp,
        dtm_modified -> Timestamp,
        value -> Nullable<Integer>,
    }
}
```

**Comment by adamhp** [comment, marked as answer]
Took me way too long to figure out that I had a trigger in my schema that had a `where` clause pointing at `.id` for these models with composite primary keys, when it needed to have been looking for the given columns in the key. Very difficult to spot in this case because it felt like it was related to the update operation but it was actually the trigger.

Double check your triggers!

**Reply by wolfymaster** [reply]
Its a year and a half later, but this just probably saved me a few more hours of trying to debug what was going on. I checked my triggers and sure enough, I made the same mistake. Thanks. :+1:
